src/Route.js

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { getProp, isControlledProp } from './util/props';
+import withYMaps from './withYMaps';
+import * as events from './util/events';
+import applyRef from './util/ref';
+import { withParentContext } from './Context';
+
+export class Route extends React.Component {
+  constructor() {
+    super();
+    this.state = { instance: null };
+  }
+
+  componentDidMount() {
+    Route.mountObject(this.props.ymaps.route, this.props).then(instance =>
+      this.setState({ instance })
+    );
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.state.instance !== null) {
+      Route.updateObject(this.state.instance, prevProps, this.props);
+    }
+  }
+
+  componentWillUnmount() {
+    Route.unmountObject(this.state.instance, this.props);
+  }
+
+  render() {
+    return null;
+  }
+
+  static mountObject(Route, props) {
+    const { instanceRef, parent, _events } = events.separateEvents(props);
+
+    const points = getProp(props, 'points');
+    const params = getProp(props, 'params');
+
+    return new Promise((resolve, reject) => {
+      Route(points, params).then(instance => {
+        if (
+          parent &&
+          parent.geoObjects &&
+          typeof parent.geoObjects.add === 'function'
+        ) {
+          parent.geoObjects.add(instance);
+        } else {
+          throw new Error(`No parent found to mount ${props.name}`);
+        }
+
+        applyRef(null, instanceRef, instance);
```

Comment:
Don't think that this will do much difference but just to be more consistent with the other components can you move `applyRef` after adding events?